### PR TITLE
Add minimal Electron dependencies

### LINKED ISSUE #89: More specific dependencies
I noticed today that the dependency list is really short, compared to [the amount of logic needed for `electron-installer-debian`](https://github.com/electron-userland/electron-installer-debian/blob/v1.0.0/src/dependencies.js). Not knowing much about the RedHat side of Linux, is this normal? Should it look more like `electron-installer-debian`?

CC: @fcastilloec

### PR DESCRIPTION
**Pull request by malept**
Fixes #89.

### CODE REVIEW

**Comment by malept** [comment]
Evidently I need to figure out how to put a newer version of RPM onto the CI worker.

**Comment by malept** [comment]
> The best I can think of is checking which version of `rpm` is being used and produce a warning saying that trash dependencies won't be added to the required packages, which will let the module build the rpm. Not all apps using electron are going to need to manipulate files and use the trash dependencies, so not having them explicitly mention won't affect the majority of projects out there.

:+1: this is what I did.

> will `electron-forge` be affected?

Yes. That's the main reason why I'm backporting a "good" version of RPM. Forge always runs against the latest stable version of Electron,and I'd rather not have the warning in CI. I've successfully built RPM 4.13.0.1 on xenial, I just need to remember how to publish a PPA.

File: README.md [hunk 1]
```diff
@@ -272,7 +275,7 @@ Machine architecture the package is targeted to, used to set the `--target` opti
 
 #### options.requires
 Type: `Array[String]`
-Default: `["lsb", "libXScrnSaver"]`
+Default: The minimum list of packages needed for Electron to run
 
 Packages that are required when the program starts, used in the [`Requires` field of the `spec` file](https://fedoraproject.org/wiki/How_to_create_an_RPM_package#Creating_a_SPEC_file).
```
Comment (fcastilloec) [inline README.md]: We need this line which is found in `-debian` 
>All user dependencies will be appended to the Default array of dependencies and any duplicates will be removed.

File: src/dependencies.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
+'use strict'
+
+const dependencies = require('electron-installer-common/src/dependencies')
```
Comment (fcastilloec) [inline src/dependencies.js]: We need to export this function on the `-common` module. I also think that the `index.js` on the `-common` module needs some cleaning. Probably move the definitions of functions to another file and keep `index.js` solely for importing/exporting of needed functions.

File: src/installer.js [hunk 3]
```diff
@@ -7,8 +7,10 @@ const dependencies = require('electron-installer-common/src/dependencies')
 const fs = require('fs-extra')
 const nodeify = require('nodeify')
 const path = require('path')
+const readElectronVersion = require('electron-installer-common/src/readelectronversion')
```
Comment (fcastilloec) [inline src/installer.js]: This function also needs exporting

**Comment by malept** [comment]
> Let me know when you've done it, I would love to use it on some of my other projects

I don't remember if I updated on this front. I couldn't get a PPA working for whatever reason (no error messages on the Launchpad side), so I ended up making an [Ubuntu repo on bintray](https://bintray.com/electronuserland/rpm-xenial).

**Comment by fcastilloec** [comment]
We haven't updated it on this module yet. Travis is still using `rpm` from the Ubuntu repos. We need to add your bintray repo to Travis to stop the warning messages.
It'll be great if we had a PPA, but any repo is fine. I also had some trouble with Launchpad so I host PPA on Github on the `gh-pages` branch. For any outside repo, we also need to add gpg keys for it.

**Comment by malept** [comment]
> We need to add your bintray repo to Travis to stop the warning messages.

I don't think it's necessary. It's not causing the builds to fail, and we're already running with `DEBUG=electron-installer-redhat`, which is a lot of noise to begin with.

> For any outside repo, we also need to add gpg keys for it.

Yes, see: https://github.com/electron-userland/electron-forge-container/commit/5847481aa37f6b34550256f9fd6e6a6e2bce8375

**Pull request merged**